src/App.tsx

```diff
@@ -1,223 +1,139 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  FormEvent,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { createTodo, deleteTodo, getTodos } from './api/todos';
+import { ErrorNotification } from
+  './components/ErrorNotification/ErrorNotification';
+import { getFilteredTodo } from './components/FilteredTodos/getFilteredTodos';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { FilterType } from './types/FilterBy';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [title, setTitle] = useState('');
+  const [selectedIds, setSelectedIds] = useState<number[]>([]);
+  const [isAdding, setIsAdding] = useState(false);
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    const getTodosFromServer = async (userId: number) => {
+      try {
+        const receivedTodos = await getTodos(userId);
+
+        setTodos(receivedTodos);
+      } catch (error) {
+        setErrorMessage(`${error}`);
+      }
+    };
+
+    if (!user) {
+      return;
     }
+
+    getTodosFromServer(user.id);
   }, []);
 
+  const filterTodoBy = getFilteredTodo(filterType, todos);
+
+  const newTodo = useCallback(async (event: FormEvent) => {
+    event.preventDefault();
+    if (!title || !user) {
+      setErrorMessage('Title cannot be empty');
+
+      return;
+    }
+
+    setIsAdding(true);
+
+    try {
+      const postTodo = await createTodo(title, user.id);
+
+      setTodos((prevTodos) => [...prevTodos, postTodo]);
+    } catch {
+      setErrorMessage('Unable to add a todo');
+    }
+
+    setIsAdding(false);
```

Comment:
try{}catch{} work async so your setIsAdding changes finishing in 2-5ms. add finaly {}